### PR TITLE
Add Python 3 compatibility to bin/cordova_plist_to_config_xml

### PR DESCRIPTION
**Pull request by cclauss**
These changes should be compatible with both Python 2 and Python 3.

### Checklist

- [ ] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

**Comment by brodybits** [comment]
Thanks for the contribution, I just requested a review from some iOS experts. My apologies for the delay.

File: bin/cordova_plist_to_config_xml [hunk 1]
```diff
@@ -101,7 +106,7 @@ def UpdateProjectFile(path):
     if 'Cordova.plist' in line:
       line = line.replace('Cordova.plist', 'config.xml')
       line = line.replace('lastKnownFileType = text.plist.xml', 'lastKnownFileType = text.xml')
-    print line,
+    print(line, end='')
```
Comment (DavidM42) [inline bin/cordova_plist_to_config_xml]: Why is it important to overwrite the newline character at the end of print here? Does this need to print in one line? Else `end=''` could be left out

**Comment by NiklasMerz** [comment]
> lgtm. Asked if `end=''` really is needed but that's non breaking

If have no idea about python, that's why I got someone to review it who has :-)

File: bin/cordova_plist_to_config_xml (hunk 1 above)
Comment (cclauss) [reply bin/cordova_plist_to_config_xml]: Did you forget this line?  `from __future__ import print_function`

Comment (DavidM42) [reply bin/cordova_plist_to_config_xml]: Yeah, you're right. Never worked with that before

**Comment by erisu** [comment]
Just as a side-note, this file seems to be used only for the Platform-Centered workflow.

There are multiple ways of going about creating a Platform-Centered project. One is a fresh project from Xcode. This means the Cordova's Xcode based project is not used and therefore does not have the `config.xml` file.

IMO, I think we do not have a lot of resources to focus on the Platform-Center workflow. I think it may have been discussed to consolidate our focus only on the CLI workflow and potentially dropping Platform-Center workflow.

If the changes work, we could just push it through and not spend too much time on this.

Just My Opinion...

**Comment by brodybits** [comment]
Agreed @erisu, let's get this merged.

I just raised #791 to avoid losing track of improving the [platform-centered workflow ([1])](https://cordova.apache.org/docs/en/latest/guide/platforms/ios/#platform-centered-workflow) and hopefully alleviate the need for Python (someday). Definitely very unfortunate that we cannot afford the resources to make this kind of improvement right now.

[1] https://cordova.apache.org/docs/en/latest/guide/platforms/ios/#platform-centered-workflow

**Comment by brodybits** [comment]
I think we want to drop this functionality in the near future, as I proposed in <https://github.com/apache/cordova/issues/198>. I wonder if we can get this fix into a minor release somehow. What do you think @erisu?

@erisu let's please get this merged in the next 12-24 hours if there are no objections, I do think it would be good to get this solution committed even if we do not want to support it too much longer.

File: bin/cordova_plist_to_config_xml [hunk 2]
```diff
@@ -106,7 +106,7 @@ def UpdateProjectFile(path):
     if 'Cordova.plist' in line:
       line = line.replace('Cordova.plist', 'config.xml')
       line = line.replace('lastKnownFileType = text.plist.xml', 'lastKnownFileType = text.xml')
-    print(line, end='')
+    print(line, end=' ')
```
Comment (brodybits) [inline bin/cordova_plist_to_config_xml]: Why add a space here?

Comment (cclauss) [reply bin/cordova_plist_to_config_xml]: To put a space between each line.

Comment (brodybits) [reply bin/cordova_plist_to_config_xml]: I don't really understand (did not work with Python for about 10 years), hope someone else can review it. We do appreciate your efforts to get this whole Python compatibility issue fixed.

Comment (cclauss) [reply bin/cordova_plist_to_config_xml]: Which do we want?
```
>>> for char in "abcdefg":
...     print(char, end="")  # --> abcdefg
    -- or --
>>> for char in "abcdefg":
...     print(char, end="")  # --> a b c d e f g
```

**Comment by knight9999** [comment]
I tried this PR with Python 3.6.4. But I got the following error.
```
TypeError: string argument expected, got 'bytes'
```

I feel it is better to use `BytesIO` with `ElementTree.write` in place of `StringIO` as follows.
```
diff --git a/bin/cordova_plist_to_config_xml b/bin/cordova_plist_to_config_xml
index f6c47505..94fe3371 100755
--- a/bin/cordova_plist_to_config_xml
+++ b/bin/cordova_plist_to_config_xml
@@ -38,6 +38,7 @@ try:  # Python 2
 except ImportError:  # Python 3
   from io import StringIO
 
+from io import BytesIO
 
 def Usage():
   sys.stderr.write(__doc__)
@@ -93,10 +94,11 @@ def ConvertPlist(src_path, dst_path):
     root.append(ElementTree.Element('access', attrib={'origin':value}))
 
   tree = ElementTree.ElementTree(root)
-  s = StringIO()
+  # s = StringIO()
+  s = BytesIO()
   tree.write(s, encoding='UTF-8')
   mini_dom = minidom.parseString(s.getvalue())
-  with open(dst_path, 'w') as out:
+  with open(dst_path, 'wb') as out:
     out.write(mini_dom.toprettyxml(encoding='UTF-8'))
```

**Pull request merged**